README.md
Blargg!
=======

**yet another** Django-powered blog. The goal of this app is to provide a
minimalistic (in both number of features and code) tool for publishing a
mostly-text blog. Shameless plug: [bradmontgomery.net](https://bradmontgomery.net) for an example.

This app provides a fairly simple `Entry` model, and a light-weight `Tag` model.

Features
--------

* Support for `Site`
* Simple content tagging
* RSS and Atom Feeds
* Sitemap support
* Automatic cross-posting to Blogger (see below)
* Support authoring content in reStructured Text (requires
  [docutils](https://pypi.python.org/pypi/docutils), markdown (requires
  [Markdown](https://pypi.python.org/pypi/Markdown) or plain old HTML.

Installation
------------

1. Install with `pip install django-blargg`
2. Add `blargg` to your `INSTALLED_APPS`
3. Configure your Root URLconf: `path('blog/', include('blargg.urls', namespace='blargg'))`
4. Customize your templates. There are some sample templates under
   `blargg/templates/blargg`, but you'll want to override these in your project.
5. Add a `BLARGG` setting dict that contains the title and description to be
   used in your blog's RSS feeds (see below).
6. (Optinally) enable Mail2Blogger (see below)


### Sample BLARGG settings

    BLARGG = {
        'title': 'Awesome blog',
        'description': 'Entries from an awesome blog',
    }


Mail2Blogger Support
--------------------

This app supports cross-posting to Blogger via email. By default, this setting
is disabled. To enable Mail2Blogger, include the following settings::

    BLARGG = {
        'title': ...,
        'description': ...,
        'mail2blogger': True,
        'mail2blogger_email': 'username.SECRET@blogger.com',
    }

To set your Mail2Blogger email address, view your
[Mobile and email](http://www.blogger.com) settings. Entries that are
cross-posted to Blogger will automatically include a link back to the original
site (provided that you've got the `django.contrib.sites` app installed and
configured correctly).

**Warning**: *mail2blogger* appears to remove extra whitespace from your content
which breaks things in a `<pre>` or `<code>` blocks. If you know how to
prevent this, [please let me know](https://github.com/bradmontgomery/django-blargg/issues/3).

License
-------

This code is distributed under the terms of the MIT license. See the `LICENSE.txt` file.


Contributing
------------

Feel free to report any issues or (better yet) pull requests for bug fixes!
There's a `requirements.txt` file in this repo to help you get all the
dependencies you'll likely need (for python3).


Why the name "Blargg"?
----------------------

From http://nintendo.wikia.com/wiki/Blargg:

    A Blargg is a large red dragon that lives in lava and appears in Super Mario
    World. When Mario goes across a lava pit on a pile of skulls occasionally
    the eyes of a Blargg will appear and then itself will try to eat Mario. They
    are found mostly in the Vanilla Dome. Strangely, they have one lazy eye.

Other than that, *Blargg* is really fun to say out loud. Especially if you use
a *Pirate Voice*. Go ahead, try it!

**Blargg**.



HISTORY.md
History
-------


## 0.7.0 (2019-01-03)

- Django 2 support (2.1.4)
- Added `on_delete` options for FK fields.
- Added title/description settings for feeds.
- Converted `url` use to `path` function in urlconf.
- Removed old south migrations.
- Include words to ignore when generating word stats.
- Add `Entry.tag_list` method that returns a sorted list of tag names


## 0.6.0 (2016-12-13)

- Django 1.8/1.9 support (with django migrations)
- Python 3 support
- Removing old, proably worthless management command
- Support for pluggable User models.
- Support for Markdown
- Adding to pypi


## 0.5.2 (2016-01-04)

- fixes for tests
- correctly count words on EntryYearArchive View

## 0.5.1 (2016-01-03)

- includes tags in the TaggedEntryListView

## 0.5.0 (2015-05-03)

- Removed dates from URLs
- Admin improvements


## 0.4.3 and earlier (2014-07-16)

- haphazardly built this thing


blargg/migrations/0001_initial.py
# -*- coding: utf-8 -*-
# Generated by Django 1.9.1 on 2016-01-07 23:16
from __future__ import unicode_literals

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('sites', '0002_alter_domain_unique'),
    ]

    operations = [
        migrations.CreateModel(
            name='Entry',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('title', models.CharField(max_length=256)),
                ('raw_content', models.TextField(help_text='Content entered by the author.')),
                ('content_format', models.CharField(choices=[('html', 'HTML'), ('rst', 'reStructured Text'), ('md', 'Markdown (not yet supported)')], max_length=4)),
                ('rendered_content', models.TextField(editable=False)),
                ('published', models.BooleanField(default=False, help_text='Show this entry to the public')),
                ('slug', models.SlugField(help_text='A slug/url used to identify this entry.', max_length=255, unique=True)),
                ('date_slug', models.SlugField(editable=False, help_text='A slug/url used to identify this entry; also includes the date on which the entry was published.', max_length=255, unique=True)),
                ('tag_string', models.TextField(blank=True, help_text='A comma-separated list of tags.')),
                ('published_on', models.DateTimeField(blank=True, editable=False, null=True)),
                ('updated_on', models.DateTimeField(auto_now=True)),
                ('created_on', models.DateTimeField(auto_now_add=True)),
                ('author', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
                ('site', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='sites.Site')),
            ],
            options={
                'verbose_name': 'Entry',
                'get_latest_by': 'published_on',
                'ordering': ['-published_on', 'title'],
                'verbose_name_plural': 'Entries',
            },
        ),
        migrations.CreateModel(
            name='Tag',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=256)),
                ('slug', models.SlugField(editable=False, max_length=256, unique=True)),
            ],
            options={
                'verbose_name': 'Tag',
                'ordering': ['name'],
                'verbose_name_plural': 'Tags',
            },
        ),
        migrations.AddField(
            model_name='entry',
            name='tags',
            field=models.ManyToManyField(editable=False, to='blargg.Tag'),
        ),
    ]
